src/test/java/org/opentripplanner/middleware/testUtils/OtpTestUtils.java

```diff
@@ -0,0 +1,206 @@
+            }
+            LOG.info("Returning mock response at index {}", mockResponseIndex);
+            // send back response and increment response index
+            String responseBody = mapper.writeValueAsString(mockResponses.get(mockResponseIndex));
+            mockResponseIndex++;
+            return responseBody;
+        }
+
+        // mocks not setup, simply return from a file every time
+        LOG.info("Returning default mock response from file");
+        OtpDispatcherResponse otpDispatcherResponse = new OtpDispatcherResponse();
+        otpDispatcherResponse.responseBody = CommonTestUtils.getResourceFileContentsAsString(
+            "otp/response/planResponse.json"
+        );
+        return otpDispatcherResponse.responseBody;
+    }
+
+    /**
+     * Provide a defined list of mock Otp Responses.
+     */
+    public static void setupOtpMocks(List<OtpResponse> responses) {
+        mockResponses = responses;
+        mockResponseIndex = 0;
+        LOG.info("Added {} Otp mocks", responses.size());
+    }
+
+    /**
+     * Helper method to reset the mocks and also make sure that the expected amount of requests were served
+     */
+    public static void resetOtpMocks() {
+        if (mockResponseIndex > -1) {
+            if (mockResponseIndex != mockResponses.size()) {
+                throw new RuntimeException(
+                    String.format(
+                        "Unexpected amount of mocked OTP responses was used. Expected=%d, Actual=%d",
+                        mockResponses.size(),
+                        mockResponseIndex
+                    )
+                );
+            }
+            LOG.info("Reset OTP mocks!");
+        }
+        mockResponses = Collections.emptyList();
+        mockResponseIndex = -1;
+    }
+
+    /**
+     * Submit plan query to OTP server and return the response.
+     */
+    public static OtpDispatcherResponse sendSamplePlanRequest() {
+        // Submit a query to the OTP server.
+        // From P&R to Downtown Orlando
+        return OtpDispatcher.sendOtpPlanRequest(
+            "28.45119,-81.36818",
+            "28.54834,-81.37745",
+            "08:35"
+        );
+    }
+
+    /**
+     * Get successful plan response from file for creating trip summaries.
+     */
+    public static OtpResponse getPlanResponse() throws IOException {
+        return CommonTestUtils.getResourceFileContentsAsJSON(
+            responseResourceFilePath + "planResponse.json",
+            OtpResponse.class
+        );
+    }
+
+    /**
+     * Get error plan response from file for creating trip summaries.
+     */
+    public static OtpResponse getPlanErrorResponse() throws IOException {
+        return CommonTestUtils.getResourceFileContentsAsJSON(
+            responseResourceFilePath + "planErrorResponse.json",
+            OtpResponse.class
+        );
+    }
+
+    static Itinerary createItinerary() {
```

Comment:
This is used only in `PersistenceTestUtils`, so I would move this test to that class.

Reply:
Moved in 695f54c4ffb08e23144a496d611b999c93c86764.